src/WebJobs.Script.Grpc/Channel/GrpcWorkerChannel.cs

```diff
@@ -402,9 +403,22 @@ internal void WorkerInitResponse(GrpcEvent initEvent)
                 ScriptHost.IsFunctionDataCacheEnabled = false;
             }
 
+            if (_environment.IsApplicationInsightsAgentEnabled() ||
```

Comment:
Question about the `||` condition here. This means the app setting cannot override the worker behavior (_worker sent a capability indicating ai logging enabled_). But customer wants to override it for some reason without redeploying worker payload and this _or_ condition will not allow that. Is that intentional behavior?

Reply:
Yeah, it was intentional. My thinking was that if either of these are set, we need to respect that and use it. In reality, most people won't even know about this and it's up to the workers/infrastructure to set these.